### PR TITLE
Pruning schedule supports fpgm

### PR DESCRIPTION
**Pull request by chicm-ms**

### CODE REVIEW

File: nni/algorithms/compression/pytorch/pruning/admm_pruner.py [hunk 1]
```diff
@@ -106,31 +106,14 @@ def _projection(self, weight, sparsity):
...
-            num_prune = int(filters * sparsity)
-            if filters < 2 or num_prune < 1:
-                mask_weight = torch.ones(weight.size()).type_as(weight).detach()
-            else:
-                w_abs_structured = w_abs.view(filters, -1).sum(dim=1)
-                threshold = torch.topk(w_abs_structured.view(-1), num_prune, largest=False)[0].max()
-                mask_weight = torch.gt(w_abs_structured, threshold)[:, None, None, None].expand_as(weight).type_as(weight)
-
-        return weight.data.mul(mask_weight)
+        return weight.data.mul(self.masker.calc_mask(sparsity, wrapper)['weight_mask'])
```
Comment (suiguoxin) [inline nni/algorithms/compression/pytorch/pruning/admm_pruner.py:116]: It seems that `self.masker.calc_mask(sparsity, wrapper)` will calculate mask based on `wrapper.module.weight` instead of the variable `weight`. Could you please check that ?

Comment (chicm-ms) [reply nni/algorithms/compression/pytorch/pruning/admm_pruner.py:116]: good catch,

Comment (chicm-ms) [reply nni/algorithms/compression/pytorch/pruning/admm_pruner.py]: Fixed

File: nni/algorithms/compression/pytorch/pruning/admm_pruner.py [hunk 2]
```diff
@@ -113,7 +114,9 @@ def _projection(self, weight, sparsity, wrapper):
         tensor
             the projected matrix
         '''
-        return weight.data.mul(self.masker.calc_mask(sparsity, wrapper)['weight_mask'])
+        wrapper_copy = copy.deepcopy(wrapper)
```
Comment (colorjam) [inline nni/algorithms/compression/pytorch/pruning/admm_pruner.py:117]: Why deepcopy here?

File: nni/algorithms/compression/pytorch/pruning/admm_pruner.py [hunk 3]
```diff
@@ -106,31 +107,16 @@ def _projection(self, weight, sparsity):
             original matrix
         sparsity : float
             the ratio of parameters which need to be set to zero
-
+        wrapper: PrunerModuleWrapper
+            layer wrapper of this layer
```
Comment (colorjam) [inline nni/algorithms/compression/pytorch/pruning/admm_pruner.py:111]: Add a blank line here

**Review by suiguoxin — APPROVED** [review]

File: nni/algorithms/compression/pytorch/pruning/admm_pruner.py (hunk 2 above)
Comment (chicm-ms) [reply nni/algorithms/compression/pytorch/pruning/admm_pruner.py:118]: It uses copied wrapper as a temporary variable to calculate mask for `weight` tensor to fix the problem that Guoxin mentioned.

File: nni/algorithms/compression/pytorch/pruning/admm_pruner.py (hunk 3 above)
Comment (chicm-ms) [reply nni/algorithms/compression/pytorch/pruning/admm_pruner.py:111]: Thanks, fixed.

**Review by colorjam — APPROVED** [review]

**Review by liuzhe-lz — APPROVED** [review]

File: nni/algorithms/compression/pytorch/pruning/net_adapt_pruner.py [hunk 4]
```diff
@@ -149,7 +149,7 @@ def _update_config_list(self, config_list, op_name, sparsity):
                 return config_list_updated
 
         # if op_name is not in self._config_list_generated, create a new json item
-        if self._base_algo in ['l1', 'l2']:
+        if self._base_algo in ['l1', 'l2', 'fpgm']:
```
Comment (QuanluZhang) [inline nni/algorithms/compression/pytorch/pruning/net_adapt_pruner.py:152]: a little confused. why ADMMPruner need to update `_projection` function, but net_adapter_pruner does not need such modification?

**Review by QuanluZhang — APPROVED** [review]
